lib/elixir/lib/macro.ex

```diff
@@ -370,6 +370,27 @@ defmodule Macro do
       iex> Macro.escape({:unquote, [], [1]}, unquote: true)
       1
 
+  ## Comparison to `Kernel.quote/2`
+
+  The `escape/2` function is sometimes confused with `Kernel.quote/2`,
+  because the above examples behave the same with both. The key difference is
+  best illustrated when the value to escape is stored in a variable.
+
+      iex> quote do: {:a, :b, :c}
+      {:{}, [], [:a, :b, :c]}
+
+      iex> value = {:a, :b, :c}
+      iex> Macro.escape(value)
+      {:{}, [], [:a, :b, :c]}
+
+      iex> quote do: value
+      {:value, [], __MODULE__}
+
+      iex> value = {:a, :b, :c}
```

Comment:
@whatyouhide updated! I'm seeing missing variable errors when I don't bind the value for each example. I forget how doctests are isolated from one another, but perhaps it isn't surprising? Let me know what you think!

Reply:
Yeah bind it on every example. That's very much fine :)